futurenhs.api/FutureNHS.Api/Services/GroupService.cs

```diff
@@ -355,5 +356,126 @@ private async Task<(GroupDto, ImageDto?)> UploadGroupImageMultipartContent(Group
             return mediaType.Encoding;
         }
 
+        public async Task<(uint, IEnumerable<GroupMember>)> GetGroupMembersAsync(Guid userId, string slug, uint offset, uint limit, string sort, CancellationToken cancellationToken)
+        {
+            if (Guid.Empty == userId) throw new ArgumentOutOfRangeException(nameof(userId));
+            if (string.IsNullOrEmpty(slug)) throw new ArgumentOutOfRangeException(nameof(slug));
+
+            var userCanPerformAction = await _permissionsService.UserCanPerformActionAsync(userId, slug, GroupViewRole, cancellationToken);
+            if (!userCanPerformAction)
+            {
+                _logger.LogError($"Error: GetGroupMembersAsync - User:{0} does not have permission to get members of this group:{1}(slug)", userId, slug);
+                throw new ForbiddenException($"Error: User does not have access");
+            }
+
+            var groupMembers = await _groupDataProvider.GetGroupMembersAsync(slug, offset, limit, sort, cancellationToken);
+
+            return groupMembers;
+        }
+
+        public async Task<(uint, IEnumerable<PendingGroupMember>)> GetPendingGroupMembersAsync(Guid userId, string slug, uint offset, uint limit, string sort, CancellationToken cancellationToken)
+        {
+            if (Guid.Empty == userId) throw new ArgumentOutOfRangeException(nameof(userId));
+            if (string.IsNullOrEmpty(slug)) throw new ArgumentOutOfRangeException(nameof(slug));
+
+            var userCanPerformAction = await _permissionsService.UserCanPerformActionAsync(userId, slug, GroupViewRole, cancellationToken);
+            if (!userCanPerformAction)
+            {
+                _logger.LogError($"Error: GetPendingGroupMembersAsync - User:{0} does not have permission to get pending group members of this group:{1}(slug)", userId, slug);
+                throw new ForbiddenException($"Error: User does not have access");
+            }
+
+            var pendingGroupMembers = await _groupDataProvider.GetPendingGroupMembersAsync(slug, offset, limit, sort, cancellationToken);
+
+            return pendingGroupMembers;
+        }
+
+        public async Task<Group?> GetGroupAsync(string slug, Guid userId, CancellationToken cancellationToken)
+        {
+            if (Guid.Empty == userId) throw new ArgumentOutOfRangeException(nameof(userId));
+            if (string.IsNullOrEmpty(slug)) throw new ArgumentOutOfRangeException(nameof(slug));
+
+            var userCanPerformAction = await _permissionsService.UserCanPerformActionAsync(userId, slug, GroupViewRole, cancellationToken);
+            if (!userCanPerformAction)
+            {
+                _logger.LogError($"Error: GetGroupMembersAsync - User:{0} does not have permission to get group:{1}(slug)", userId, slug);
```

Comment:
Should be `GetGroupAsync`, few other log errors referencing the wrong method in here also